python/print.hpp

```diff
@@ -0,0 +1,134 @@
+#pragma once
+
+#include <cxxabi.h>
+
+/*
+ * Utility code for generating string representations of types for use in Python
+ * wrappers.
+ */
+
+#include <string>
+#include <sstream>
+
+#include <domain_decomposition.hpp>
+#include <hardware/node_info.hpp>
+#include <profiling/meter_manager.hpp>
+#include <recipe.hpp>
+#include <rss_cell.hpp>
+
```

Comment:
If these are all going to be included in the main source anyhow, I think it would be clearer to have the definition of the to-string functions next to the point of their use in the adaptors.

Reply:
I have them in a seperate file because there is enough code in the `arb.cpp` file, without cluttering it with the print functions. I figured that a good compromise is to have all "stringify" code in one place.